cloudferrylib/os/actions/recreate_boot_image.py

```diff
@@ -0,0 +1,79 @@
+            #img_id value is used as a name of new image because we can't get name of deleted image
+            for image_id_src, gl_image in images_info['images'].iteritems():
+                if image_id_src == img_id and not gl_image['image']:
+                    diff = gl_image['meta']['instance'][0]['diff']['path_src']
+                    img_src_host = gl_image['meta']['instance'][0]['diff']['host_src']
+                    if img_src_host == src_host:
+                        qemu_img_src = self.src_cloud.qemu_img
+                        base = qemu_img_src.detect_backing_file(diff, src_host)
+                        src_runner = remote_runner.RemoteRunner(src_host,
+                                                                src_user,
+                                                                password=src_password,
+                                                                sudo=True)
+                        dst_runner = remote_runner.RemoteRunner(dst_host,
+                                                                dst_user,
+                                                                password=dst_password,
+                                                                sudo=True)
+                        with files.RemoteTempDir(src_runner) as src_tmp_dir, files.RemoteTempDir(dst_runner) as dst_tmp_dir:
+                            diff_name = 'diff'
+                            base_name = 'base'
+                            diff_file = os.path.join(src_tmp_dir, diff_name)
+                            base_file = os.path.join(src_tmp_dir, base_name)
+                            dst_base_file = os.path.join(dst_tmp_dir, base_name)
+                            src_runner.run('cp {} {}'.format(diff, diff_file))
+                            if base:
+                                src_runner.run('cp {} {}'.format(base, base_file))
+                                qemu_img_src.diff_rebase(base_file, diff_file, src_host)
+                                qemu_img_src.diff_commit(src_tmp_dir, diff_name, src_host)
+                                verified_file_copy(src_runner, dst_runner, dst_user, base_file, dst_base_file, dst_host, 1)
+                            else:
+                                verified_file_copy(src_runner, dst_runner, dst_user, diff_file, dst_base_file, dst_host, 1)
+                            image_resource = self.dst_cloud.resources[utils.IMAGE_RESOURCE]
+                            new_img_id = image_resource.glance_img_create(dst_runner, img_id, 'qcow2', dst_base_file)
+                            gl_image['image']['id'] = new_img_id
+                            gl_image['image']['resource'] = None
+                            gl_image['image']['checksum'] = remote_md5_sum(dst_runner, dst_base_file)
+                            gl_image['image']['name'] = img_id
+                    else:
+                        LOG.error('Wrong host in the image info')
```

Comment:
1. The error message is not descriptive to the user. Please change.
2. There is no exception raised, so I'm assuming the error is not critical, so let's change it to warning instead.